scan-command/src/test/java/io/ballerina/scan/internal/Rule005Test.java

```diff
@@ -0,0 +1,35 @@
+package io.ballerina.scan.internal;
+
+import io.ballerina.projects.Document;
+import io.ballerina.scan.Issue;
+import io.ballerina.scan.RuleKind;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.util.List;
+
+public class Rule005Test extends StaticCodeAnalyzerTest {
+    private static final String UNUSED_CLASS_PRIVATE_FIELDS = "Unused class private fields";
+
+    @Test(description = "test checkpanic analyzer")
+    void testUnusedClassFieldsAnalyzer() {
+        String documentName = "rule005_unused_class_fields.bal";
+        Document document = loadDocument(documentName);
+        ScannerContextImpl scannerContext = new ScannerContextImpl(List.of(CoreRule.UNUSED_CLASS_FIELDS.rule()));
+        StaticCodeAnalyzer staticCodeAnalyzer = new StaticCodeAnalyzer(document, scannerContext,
+                document.module().getCompilation().getSemanticModel());
+        staticCodeAnalyzer.analyze();
+        List<Issue> issues = scannerContext.getReporter().getIssues();
+        Assert.assertEquals(issues.size(), 5);
```

Comment:
Why not use a variable to assert total instead, like we do with compiler tests? That way we know we assert everything.

Reply:
This will improve the readability and this will also checks the relevent fields in the rule as well

Reply:
> checks the relevent fields in the rule as well

Can you please elaborate?

Even in terms of readability, the number of errors doesn't really add value does it? If we follow the approach of using a variable that gets incremented with the assertion of each rule violation, we make sure that all the violations are asserted and consciously verified before making a change here too.

Reply:
Since this apply to all the rules, will create a separate issue to track

Reply:
Created https://github.com/ballerina-platform/static-code-analysis-tool/issues/47